src/shim/kernel.rs

```diff
@@ -8,77 +9,52 @@ use fvm4::kernel::SyscallError as E4;
 use fvm_shared2::error::ErrorNumber as N2;
 use fvm_shared3::error::ErrorNumber as N3;
 use fvm_shared4::error::ErrorNumber as N4;
-use itertools::Either;
+use num_traits::FromPrimitive;
 use std::fmt;
 
 macro_rules! error_number {
     ($($variant:ident),* $(,)?) => {
         #[derive(Debug, Clone)]
         #[non_exhaustive]
         pub enum ErrorNumber {
-            $($variant,)*
-            Unknown(UnknownErrorNumber),
+            $($variant(u32),)*
```

Comment:
The enum variants should not take a `u32`. They statically map to a single number.

Reply:
I removed them. Where I am still unsure of is the need of the `from` function.